src/ethereum_spec_tools/evm_tools/t8n/t8n_types.py

```diff
@@ -0,0 +1,348 @@
+                    # unsupported transaction type is attempted to be
+                    # signed. If a signed unsupported transaction is
+                    # provided, it will simply be rejected and the
+                    # next transaction is attempted.
+                    # See: https://github.com/ethereum/go-ethereum/issues/26861
+                    t8n.logger.error(f"Rejected transaction {idx}")
+                    raise FatalException(e)
+
+            try:
+                tx = t8n.json_to_tx(json_tx)
+
+            except Exception as e:
+                t8n.logger.warning(f"Rejected transaction {idx}: {e}")
+                self.rejected_txs[idx] = str(e)
+                continue
+
+            if t8n.is_after_fork("ethereum.berlin"):
+                transaction = t8n.fork_types.decode_transaction(tx)
+            else:
+                transaction = tx
+
+            yield idx, transaction
+
+    def sign_transaction(self, json_tx: Any) -> None:
+        """
+        Sign a transaction. This function will be invoked if a `secretKey`
+        is provided in the transaction.
+        Post spurious dragon, the transaction is signed according to EIP-155
+        if the protected flag is missing or set to true.
+        """
+        t8n = self.t8n
+        protected = json_tx.get("protected", True)
+
+        tx = t8n.json_to_tx(json_tx)
+
+        if isinstance(tx, bytes):
+            tx_decoded = t8n.fork_types.decode_transaction(tx)
+        else:
+            tx_decoded = tx
+
+        secret_key = hex_to_uint(json_tx["secretKey"][2:])
+        if t8n.is_after_fork("ethereum.berlin"):
+            Transaction = t8n.fork_types.LegacyTransaction
+        else:
+            Transaction = t8n.fork_types.Transaction
+
+        if isinstance(tx_decoded, Transaction):
+            if t8n.is_after_fork("ethereum.spurious_dragon"):
+                if protected:
+                    signing_hash = t8n.fork.signing_hash_155(tx_decoded)
+                    v_addend = 37  # Assuming chain_id = 1
+                else:
+                    signing_hash = t8n.fork.signing_hash_pre155(tx_decoded)
+                    v_addend = 27
+            else:
+                signing_hash = t8n.fork.signing_hash(tx_decoded)
+                v_addend = 27
+        elif isinstance(tx_decoded, t8n.fork_types.AccessListTransaction):
+            signing_hash = t8n.fork.signing_hash_2930(tx_decoded)
+            v_addend = 0
+        elif isinstance(tx_decoded, t8n.fork_types.FeeMarketTransaction):
+            signing_hash = t8n.fork.signing_hash_1559(tx_decoded)
+            v_addend = 0
+        else:
+            raise FatalException("Unknown transaction type")
+
+        r, s, y = secp256k1_sign(signing_hash, secret_key)
+        json_tx["r"] = hex(r)
+        json_tx["s"] = hex(s)
+        json_tx["v"] = hex(y + v_addend)
+
+
+class Result:
+    """Type that represents the result of a transition execution"""
+
+    state_root: Any = None
```

Comment:
Does setting a value here work in non-dataclasses? Like it doesn't set a class-level variable? My Python is pretty limited to just what I've learned working on the specs :sweat_smile:

Reply:
Ah. Yes. Just a result of going back and forth a few times between dataclasses and non-dataclasses during dev :)
Have cleaned up and defined all the attributes in the `__init__` function for non-dataclasses.